examples/basic_usage.cc

```
@@ -0,0 +1,410 @@
+#include "FileAnalysis.hh" // Include the library header
+#include <iostream>
+#include <string>
+#include <vector>
+#include <sstream> // For string stream
+#include <iomanip> // For std::fixed, std::setprecision
+#include <map>     // For JSON-like structures (conceptually)
+#include <filesystem> // For directory iteration (C++17)
+#include <utility> // For std::pair
+
+// Helper function to convert the StatusFlags bitfield to a readable string
+std::string flagsToString(StatusFlags flags) {
```

Cppcheck (2.10-2)

[style] The function 'flagsToString' is never used.

(unusedFunction)